src/CoreBundle/private/modules/CMSTemplateEngine/CMSTemplateEngine.class.php

```diff
@@ -108,18 +111,29 @@ public function Init()
         parent::Init();
 
         $this->sPageId = $this->getPageIdFromRequest();
+
+        // TODO what about language?
+        $page = $this->getPageService()->getById($this->sPageId);
+
+        if (null === $page) {
+            throw new NotFoundHttpException(sprintf('A page with the id %s cannot be found.', $this->sPageId));
+        }
+
         $this->oTableManager = new TCMSTableEditorManager();
         $this->sTableID = TTools::GetCMSTableId('cms_tpl_page');
         $this->oTableManager->Init($this->sTableID, $this->sPageId);
+
+        //$this->oPage = $this->getPageDefAccess()->get($this->sPageId);
```

Comment:
Would not change this here. If so, it needs to be tested in deep. I'm not into the details of the difference between this new service and the GetCachedInstance method. But i would like to see this legacy call deprecated. So feel free to check if the new service is right here.

Reply:
Changed: only using $this->oPage now for everything.